### ISSUE TITLE
Modifier key triggers a non matching key event

### ISSUE THREAD

**Issue by fred-dev** [openframeworks/openFrameworks#6070]
This code in the keyPressed method:

```
keyPressed(int key){
    if (key == OF_KEY_COMMAND) {
        cout<<"got the command key" << endl;
    }
    if (key != OF_KEY_COMMAND) {
        cout<<"got another key" << endl;
    }
}
```

Will output this every time the command key is pressed
```
got the command key
got another key
```

This is true for all modifier keys, shift, command control, tested on OSX with current master

**Comment by dimitre** [comment]
I'll add a link to a forum thread here with more info
https://forum.openframeworks.cc/t/command-f-how-to-handle-it/29908/5

**Comment by danomatika** [comment]
I'm seeing the same issue as the following no longer works:

~~~
// choose modifier key based on platform
#ifdef __APPLE__
	#define MOD_KEY OF_KEY_SUPER
#else
	#define MOD_KEY OF_KEY_CONTROL
#endif

...

if(ofGetKeyPressed(MOD_KEY)) {
		switch(key) {
			case 'f':
				ofToggleFullscreen();
				return;
			case 'r':
				script.reload();
				return;
			case 'c':
				script.clear();
				return;
		}
	}
~~~

When doing CMD+f, the 'f' (102) is simply not detected. It's coming through as (6) instead.

**Comment by danomatika** [comment]
Actually, these values are the same ASCII control codes you get with the control key, ie. CTL+f -> 6.

**Comment by danomatika** [comment]
Bug found. Looks like too much bitshifting causes the CMD modifier to be treated as CTL: https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/app/ofAppGLFWWindow.cpp#L1183

I guess the Apple documentation **was** to be believed...

**Comment by danomatika** [comment]
On further inspection, most of the modifiers are currently broken on macOS:

* CTRL+f = f (102), should be (6)
* CMD+f = (6), should be f (102)
* SHIFT+f = f (120), should be F (70)
* ALT+f = ƒ (402)

This is due to incorrect values being used at https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/app/ofAppGLFWWindow.cpp#L1174.

From what I can tell, the `UCKeyTranslate()` function is pretty old and expects modifier masks from the Carbon framework. The Cocoa modifier masks, ie. NSEventModifierFlagShift (NSShiftKeyMask is deprecated), are different.

From some [JNI Carbon wrapper enums](https://java-native-access.github.io/jna/4.5.0/javadoc/constant-values.html#com.sun.jna.platform.mac.Carbon.cmdKey):
~~~
shiftKey = 512
controlKey = 4096
optionKey = 2048
cmdKey = 256
~~~

From NSEvent.h:
~~~
NSEventModifierFlagShift (was NSShiftKeyMask) = 1 << 17
NSEventModifierFlagControl (was NSControlKeyMask) = 1 << 18
NSEventModifierFlagOption (was NSAlternateKeyMask) = 1 << 19
NSEventModifierFlagCommand (was NSCommandKeyMask) = 1 << 20
~~~

So shifting right by 16 gives us:
~~~
NSEventModifierFlagShift  >> 8 = 512
NSEventModifierFlagControl >> 1024
NSEventModifierFlagOption >> 2048
NSEventModifierFlagCommand >> 4096
~~~

Shifting these values either by 8 or 16 still leads to an incorrect modifier mask. It was just luck that 16 *seemed* to work before but it clearly does not.

Using the actual Carbon values fixes the key modifier behavior:

~~~
                if (modifier & GLFW_MOD_SHIFT)
			mod_OSX |= 512; // Carbon shiftKey value
		if (modifier & GLFW_MOD_CONTROL)
			mod_OSX |= 4096; // Carbon controlKey value
		if (modifier & GLFW_MOD_ALT)
			mod_OSX |= 2048; // Carbon optionKey value
		if (modifier & GLFW_MOD_SUPER)
			mod_OSX |= 256; // Carbon cmdKey

		mod_OSX = (mod_OSX >> 8) & 0xFF;
~~~

Now I get the following (as expected):

* CTRL+f = (6)
* CMD+f = f (102)
* SHIFT+f =  F (70)
* ALT+f = ƒ (402)

**Comment by danomatika** [comment]
This is fixed via PR #6087.
